src/main/java/sparksoniq/spark/DataFrameUtils.java

```diff
@@ -0,0 +1,150 @@
+
+    public static byte[] serializeItem(Item toSerialize) {
+        Kryo kryo = KM.getOrCreateKryo();
+
+        Output output = new ByteBufferOutput(128, -1);
+        kryo.writeClassAndObject(output, toSerialize);
+        output.close();
+
+        byte[] byteArray = output.toBytes();
+
+        KM.releaseKryoInstance(kryo);
+
+        return byteArray;
+    }
+
+    public static byte[] serializeItemList(List<Item> toSerialize) {
+        Kryo kryo = KM.getOrCreateKryo();
+
+        Output output = new ByteBufferOutput(128, -1);
+        kryo.writeClassAndObject(output, toSerialize);
+        output.close();
+
+        byte[] byteArray = output.toBytes();
+
+        KM.releaseKryoInstance(kryo);
+
+        return byteArray;
+    }
+
+    public static Object deserializeByteArray(byte[] toDeserialize) {
+        Kryo kryo = KM.getOrCreateKryo();
+
+        Input input = new Input(new ByteArrayInputStream(toDeserialize));
+
+        Object obj = kryo.readClassAndObject(input);
+        input.close();
+
+        KM.releaseKryoInstance(kryo);
+
+        return obj;
+    }
+
+    public static String getUdfSQL(StructType inputSchema) {
+        String[] columnNames = inputSchema.fieldNames();
+        StringBuilder queryColumnString = new StringBuilder();
+        for (int columnIndex = 0; columnIndex < columnNames.length; columnIndex++) {
+            queryColumnString.append("`");
+            queryColumnString.append(columnNames[columnIndex]);
+            queryColumnString.append("`");
+            if (columnIndex != (columnNames.length - 1)) {
+                queryColumnString.append(",");
+            }
+        }
+        return queryColumnString.toString();
+    }
+
+    public static String getSelectSQL(StructType inputSchema, int duplicateVariableIndex) {
+        String[] columnNames = inputSchema.fieldNames();
+        StringBuilder queryColumnString = new StringBuilder();
+        for (int columnIndex = 0; columnIndex < columnNames.length; columnIndex++) {
+            if (columnIndex == duplicateVariableIndex) {
+                continue;
+            }
+            queryColumnString.append("`");
+            queryColumnString.append(columnNames[columnIndex]);
+            queryColumnString.append("`");
+            queryColumnString.append(",");
+        }
+
+        return queryColumnString.toString();
+    }
+
+
+    public static List<List<Item>> deserializeWrappedParameters (WrappedArray wrappedParameters, List<List<Item>> deserializedParams) {
+        Object[] serializedParams = (Object[]) wrappedParameters.array();
+        for (int paramIndex = 0; paramIndex < serializedParams.length; paramIndex++) {
+            Object serializedParam = serializedParams[paramIndex];
+            Object deserializedParam = deserializeByteArray((byte[])serializedParam);
+            deserializedParams.add((List<Item>) deserializedParam);
+        }
+        return deserializedParams;
```

Comment:
The return value is already sent editing the second parameter (which is perfect, i.e., the ownership of the list is on the caller to improve performance).

Consequently, there is no need to return it, as this is redundant: the method can be made void.

Reply:
This will be solved with the whereClause PR in order to reduce the rebasing effort to a minimum. For let and where clauses will be addressed.